File: fe2-android/app/src/test/unit/java/com/github/dedis/popstellar/model/network/method/message/ElectionQuestionResultTest.java
package com.github.dedis.popstellar.model.network.method.message;

import com.github.dedis.popstellar.model.network.method.message.data.election.ElectionResultQuestion;
import com.github.dedis.popstellar.model.network.method.message.data.election.QuestionResult;

import org.junit.Test;

import java.util.Collections;
import java.util.Set;

import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.MatcherAssert.assertThat;
import static org.junit.Assert.assertThrows;

public class ElectionQuestionResultTest {

  private final String questionId = "questionId";
  private final Set<QuestionResult> results =
      Collections.singleton(new QuestionResult("Candidate1", 30));
  private final ElectionResultQuestion electionQuestionResult =
      new ElectionResultQuestion(questionId, results);

  @Test
  public void electionQuestionResultGetterReturnsCorrectQuestionId() {
    assertThat(electionQuestionResult.getId(), is(questionId));
  }

  @Test
  public void electionQuestionResultGetterReturnsCorrectResults() {
    assertThat(electionQuestionResult.getResult(), is(results));
  }

  @Test
  public void resultsCantBeEmpty() {
    Set<QuestionResult> emptySet = Collections.emptySet();
    assertThrows(
        IllegalArgumentException.class, () -> new ElectionResultQuestion(questionId, emptySet));
  }
}

File: fe2-android/app/src/test/unit/java/com/github/dedis/popstellar/model/network/method/message/ElectionQuestionResultTest.kt
package com.github.dedis.popstellar.model.network.method.message

import com.github.dedis.popstellar.model.network.method.message.data.election.ElectionResultQuestion
import com.github.dedis.popstellar.model.network.method.message.data.election.QuestionResult
import org.hamcrest.CoreMatchers
import org.hamcrest.MatcherAssert
import org.junit.Test

class ElectionQuestionResultTest {
  private val validQuestionId = "questionId"
  private val validCount = 30
  private val validResults = setOf(QuestionResult("Candidate1", validCount))
  private val validElectionQuestionResult = ElectionResultQuestion(validQuestionId, validResults)
  private val emptyResultSet = emptySet<QuestionResult>()

  @Test
  fun constructorSucceedsWithValidData() {
    ElectionResultQuestion(validQuestionId, validResults)
  }

  @Test(expected = IllegalArgumentException::class)
  fun constructorFailsWhenIdIsEmpty() {
    ElectionResultQuestion("", validResults)
  }

  @Test(expected = IllegalArgumentException::class)
  fun constructorFailsWhenResultSetIsEmpty() {
    ElectionResultQuestion(validQuestionId, emptyResultSet)
  }

  @Test
  fun electionQuestionResultGetterReturnsCorrectQuestionId() {
    MatcherAssert.assertThat(validElectionQuestionResult.id, CoreMatchers.`is`(validQuestionId))
  }

  @Test
  fun electionQuestionResultGetterReturnsCorrectResults() {
    MatcherAssert.assertThat(validElectionQuestionResult.result, CoreMatchers.`is`(validResults))
  }

  @Test(expected = IllegalArgumentException::class)
  fun resultsCantBeEmpty() {
    val emptySet = emptySet<QuestionResult>()
    ElectionResultQuestion(validQuestionId, emptySet)
  }
}
